docs/blog/2019-01-04-using-react-context-api-with-gatsby/index.md

```diff
@@ -0,0 +1,273 @@
+    import React from 'react'
+    import PropTypes from 'prop-types'
+    import { StaticQuery, graphql } from 'gatsby'
+
+    import ThemeContext from '../context/ThemeContext'
+    import Header from './header'
+    import './layout.css'
+
+    const Layout = ({ children }) => (
+      <StaticQuery
+        query={graphql`
+          query SiteTitleQuery {
+            site {
+              siteMetadata {
+                title
+              }
+            }
+          }
+        `}
+        render={data => (
+          // highlight-start
+          <ThemeContext.Consumer>
+            {theme => (
+              <div className={theme.dark ? 'dark' : 'light'}>
+          // highlight-end
+                <Header siteTitle={data.site.siteMetadata.title} />
+                <div
+                  style={{
+                    margin: `0 auto`,
+                    maxWidth: 960,
+                    padding: `0px 1.0875rem 1.45rem`,
+                    paddingTop: 0,
+                  }}
+                >
+                  {children}
+                  <footer>
+                    © {new Date().getFullYear()}, Built with
+                    {` `}
+                    <a href="https://www.gatsbyjs.org">Gatsby</a>
+                  </footer>
+                </div>
+              </div>
+            )}
+          </ThemeContext.Consumer>
+        )}
+      />
+    )
+
+    Layout.propTypes = {
+      children: PropTypes.node.isRequired,
+    }
+
+    export default Layout
+```
+
+Basically the class of the wrapper div will change based on the context value of the dark variable, which we set as state in the `ThemeContext.js` file.
+
+## Adding the switch in the Header
+
+Having done the configuration, let’s add the actual switch to turn on/off the dark mode. Please modify the `header.js` code to look like this:
+```jsx:title=src/components/header.js
+    import { Link } from 'gatsby'
+    import PropTypes from 'prop-types'
+    import React from 'react'
+
+    import ThemeContext from '../context/ThemeContext'
+
+    const Header = ({ siteTitle }) => (
+      <ThemeContext.Consumer>
+        {theme => (
+          <div
+            style={{
+              background: `rebeccapurple`,
+              marginBottom: `1.45rem`,
+            }}
+          >
+            <div
+              style={{
+                margin: `0 auto`,
+                maxWidth: 960,
+                padding: `1.45rem 1.0875rem`,
+              }}
+            >
+              <h1 style={{ margin: 0 }}>
+                <Link
+                  to="/"
+                  style={{
+                    color: `white`,
+                    textDecoration: `none`,
+                  }}
+                >
+                  {siteTitle}
+                </Link>
+              </h1>
+              <button className="dark-switcher" onClick={theme.toggleDark}>
+                {theme.dark ? (
+                  <span>Light mode &#9728;</span>
```

Comment:
Maybe tweak this? I don't know what these codes mean off-hand.

Reply:
Sure, I added their sign equivalents in HTML!

Reply:
I added the signs instead of the encoded entities and the result is the same but more readable.